packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasSummary.ts

```diff
@@ -94,92 +71,72 @@ export function useGasSummary(): UseGasSummaryResult {
     []
   )
 
-  const estimateGas = useCallback(async () => {
-    if (!walletAddress) {
+  const { gasEstimates: estimateGasResult, error: gasEstimatesError } =
+    useGasEstimates({
+      txType: isDepositMode ? 'deposit' : 'withdrawal',
+      walletAddress,
+      childChainProvider,
+      parentChainProvider: isDepositMode ? parentChainProvider : undefined,
+      amount: amountBigNumber,
+      tokenParentChainAddress: token ? token.address : undefined
+    })
+
+  const estimatedParentChainGasFees = useMemo(() => {
+    if (!estimateGasResult?.estimatedParentChainGas) {
       return
     }
-
-    const estimateGasFunctionParams = {
-      amount: amountDebounced,
-      address: walletAddress,
-      l2Provider: childChainProvider
+    return calculateEstimatedParentChainGasFees(
+      estimateGasResult.estimatedParentChainGas,
+      parentChainGasPrice
+    )
+  }, [estimateGasResult, parentChainGasPrice])
+
+  const estimatedChildChainGasFees = useMemo(() => {
+    if (!estimateGasResult?.estimatedChildChainGas) {
+      return
     }
+    return calculateEstimatedChildChainGasFees(
+      estimateGasResult.estimatedChildChainGas,
+      childChainGasPrice,
+      estimateGasResult.estimatedChildChainSubmissionCost
+    )
+  }, [childChainGasPrice, estimateGasResult])
 
-    let estimateGasResult: GasEstimationResult = INITIAL_GAS_ESTIMATION_RESULT
+  useEffect(() => {
```

Comment:
Maybe a crazy idea. But do we need useEffect here? Can we have it directly in the body of the component? And worst case, we would just call `setGasSummaryStatus` multiple time with the same status

Reply:
that would cause way too many rerenders (infinite loop) so we can't do that

Reply:
Changing a string to the same value should not cause new render